woeip/apps/air_quality/models.py

```diff
@@ -1,79 +1,101 @@
+        ('PM2.5', 'PM2.5'),
+        ('μg/m3', 'μg/m3'),
+        ('latlon', 'latlon'),
+    )
     unit = models.CharField(max_length=256, choices=unit_choices,
                             help_text="Measurement unit, e.g., mg/m3, ppm, etc.")
-    device = models.ForeignKey(Device, on_delete=models.SET_NULL, blank=True, null=True)
 
     def __str__(self):
-        return f"{self.name} ({self.device.name})"
+        return f'{self.name} ({self.device.name})'
 
 
-class Session(models.Model):
-    """A single air quality outing. Can link to several SessionData, e.g., raw data files."""
-    date_collected = models.DateTimeField()
-    route = models.ForeignKey(Route, on_delete=models.SET_NULL, blank=True, null=True)
-    collected_by = models.ForeignKey(User, on_delete=models.SET_NULL, blank=True, null=True)
+class Calibration(models.Model):
+    """A calibration is performed on a sensor by a user.
+    """
+    sensor = models.ForeignKey(Sensor, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    calibrated_at = models.DateTimeField(null=True)
 
     def __str__(self):
-        return f"{self.date_collected} {self.collected_by}"
+        return f'{self.sensor.name} {self.user.username}: {self.calibrated_at}'
 
 
-class SessionData(TimeStampedModel):
-    """The raw data files generated during a session. Assumes one general and one gps file.
-    Multiple sensors can be linked to one session"""
-    sensor_file = models.FileField(upload_to='sensor_files', default="")
-    gps_file = models.FileField(upload_to='gps_files', default="")
-    sensor = models.ForeignKey(Sensor, on_delete=models.SET_NULL, blank=True, null=True)
-    session = models.ForeignKey(Session, on_delete=models.CASCADE, blank=True, null=True)
-    uploaded_by = models.ForeignKey(User, on_delete=models.SET_NULL, blank=True, null=True)
+class Collection(models.Model):
+    """A collection refers to a single data collection outing. During a
+    collection, multiple collection files may be recorded from multiple devices
+    and sensors by one or more users.
+    """
+    starts_at = models.DateTimeField()
+    ends_at = models.DateTimeField()
+    route = models.CharField(max_length=256, null=True)
 
-    class Meta:
-        unique_together = ('sensor', 'session')
 
-    def __str__(self):
-        name = op.basename(self.sensor_file.name)
-        return name
+class CollectionFile(models.Model):
+    """A collection file refers to a single file recorded during a collection
+    that captures data from a single sensor. When the file contents are
+    processed, the table is updated with the timestamp and processor version.
+    """
+    collection = models.ForeignKey(Collection, on_delete=models.CASCADE)
+    sensor = models.ForeignKey(Sensor, null=True, on_delete=models.SET_NULL)
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    file = models.FileField(upload_to='data', default='')
```

Comment:
@nbilenko this file field should be a string url instead. It's my understanding that we aren't uploading the file to our servers but rather to S3 instead

Reply:
`FileField` works with the S3 bucket and maintains the file management capabilities of Django. [Overview from _Simple is Better than Complex_](https://simpleisbetterthancomplex.com/tutorial/2017/08/01/how-to-setup-amazon-s3-in-a-django-project.html)

Reply:
Indeed. See the settings section for defining the S3 config [here](https://github.com/openoakland/woeip/blob/master/woeip/settings.py#L45).

Reply:
ah good to know, thanks! @nbilenko the PR looks great 👍  let's merge